src/core/borrower/shelf.sol

```diff
@@ -168,9 +169,15 @@ contract Shelf is DSNote, Auth, TitleOwned, Math {
     }
 
     function recover(uint loan, address usr, uint currencyAmount) public auth {
-        _repay(loan, usr, currencyAmount);
-        uint loss = pile.debt(loan);
-        pile.decDebt(loan, loss);
+        pile.accrue(loan);
+        uint loanDebt = pile.debt(loan);
+        
+        currency.transferFrom(usr, address(this), currencyAmount);
+       
+        ceiling.repay(loan, currencyAmount);
```

Comment:
Not quite sure here but i think it makes sense

Reply:
I also thought about this line. Currently nothing is happening on ceiling.repay. But maybe we introduce some logic later that also is relevant for recovery so I left it in here